src/graph.py

```diff
@@ -0,0 +1,174 @@
+            for value in self.graph_dict[val]:
+                to_return.append(value)
+            return to_return
+        except KeyError:
+            raise KeyError("No such node exists.")
+    def adjacent(self, val1, val2):
+        """Return True if edge exists, else return false."""
+        try:
+            self.graph_dict[val2]
+            for node in self.graph_dict[val1]:
+                if node[0] == val2:
+                    return True
+            else: 
+                return False
+        except KeyError:
+            raise KeyError("Value given not in graph.")
+    def depth(self, val):
+        from stack import Stack
+        seen = []
+        next_up = Stack()
+        try:
+            while True:
+                if val not in seen:
+                    seen.append(val)
+                    for i in self.graph_dict[val][::-1]:
+                        next_up.push(i)
+                if len(next_up) == 0:
+                    break
+                val = next_up.pop().value[0]
+            return seen
+        except KeyError:
+            raise KeyError('Given value does not exist.')
+    def breadth(self, val):
+        from que_ import Queue
+        seen = []
+        next_up = Queue()
+        try:
+            while True:
+                if val not in seen:
+                    seen.append(val)
+                    for i in self.graph_dict[val]:
+                        next_up.enqueue(i)
+                if next_up.size() == 0:
+                    break
+                val = next_up.dequeue().value[0]
+            return seen
+        except KeyError:
+            raise KeyError('Given value does not exist.')
+    def path(self, val1):
+        from priorityq import Priority_Q
+        the_list = self.breadth(val1)
+        the_queue = Priority_Q()
+        to_return = []
+        for i in the_list:
+            if i == val1:
+                the_queue.insert(i, self.graph_dict[i], 0)
+            else:
+                the_queue.insert(i, self.graph_dict[i])
+        while len(the_queue.heap.heap) > 0:
+            current = the_queue.pop()
+            for neighbor in current['neighbors']:
+                alt = current['dist'] + neighbor[1]
+                print('at node ', current['value'])
+                print(alt, neighbor[0])
+                the_queue.decrease_priority(neighbor[0], alt, current['value'])
+            to_return.append(current)
+        return to_return
+
+if __name__ == '__main__':
+    graphy_mcgraphface = Graph()
+    graphy_mcgraphface.add_edge('A', 'C', 1)
+    graphy_mcgraphface.add_edge('A', 'B', 4)
+    graphy_mcgraphface.add_edge('A', 'D', 6)
+    graphy_mcgraphface.add_edge('C', 'E', 1)
+    graphy_mcgraphface.add_edge('E', 'C', 1)
+    graphy_mcgraphface.add_edge('E', 'B', 1)
+    graphy_mcgraphface.add_edge('B', 'D', 1)
+    graphy_mcgraphface.add_edge('B', 'F', 1)
+    graphy_mcgraphface.add_edge('F', 'D', 10)
+    print(graphy_mcgraphface.path('A'))
```

Comment:
Why has this been updated for your shortest path? that should be on a different branch